gstd/src/async_runtime/signals.rs

```diff
@@ -51,14 +51,18 @@ impl WakeSignals {
     }
 
     pub fn register_signal(&mut self, waiting_reply_to: MessageId) {
+        let message_id = crate::msg::id();
+
         self.signals.insert(
             waiting_reply_to,
             WakeSignal {
-                message_id: crate::msg::id(),
+                message_id,
                 payload: None,
                 waker: None,
             },
         );
+
+        crate::async_runtime::locks().lock(message_id, waiting_reply_to, Default::default());
```

Comment:
Would be great to have some comment here explaining why the lock is required

Reply:
To be added later in covering with logs